runtime/tests/integration_tests.rs

```diff
@@ -60,15 +66,58 @@ fn run_rt_test(test_bin_name: Option<&str>) -> DefaultHwModel {
     model
 }
 
-fn send_mailbox_command(model: &mut DefaultHwModel, cmd_id: u32, arg_buf: &[u32]) -> Vec<u32> {
+/// Stream `msg` to Caliptra's SHA hardware
+fn stream_to_sha(model: &mut DefaultHwModel, msg: &[u32]) {
+    if model.soc_sha512_acc().lock().read().lock() {
+        panic!("sha_acc lock could not be acquired");
+    }
+
+    model
+        .soc_sha512_acc()
+        .dlen()
+        .write(|_| (msg.len() * 4).try_into().unwrap());
+
+    model
+        .soc_sha512_acc()
+        .mode()
+        .write(|w| w.mode(|w| w.sha_stream_384()));
+
+    for word in msg {
+        model.soc_sha512_acc().datain().write(|_| *word);
+    }
+
+    model.soc_sha512_acc().execute().write(|w| w.execute(true));
+
+    model.step_until(|m| m.soc_sha512_acc().status().read().valid());
+
+    let digest = model.soc_sha512_acc().digest().read();
+    for word in digest {
+        println!("0x{:x}", word);
+    }
+
+    model.soc_sha512_acc().lock().write(|w| w.lock(true)); // clear lock
+}
```

Comment:
@korran @mhatrevi when I run this on verilator, it hangs on line 91 (valid bit never gets set). I tried to poke around at the waveform to see if anything is going on, but it's hard to tell.

Anything I'm doing here look obviously wrong? (Note, I don't do anything in firmware to initialize the sha512acc peripheral, but it didn't seem like anything was necessary)

Reply:
Did some more digging: https://github.com/chipsalliance/caliptra-rtl/issues/84

I tweaked the RTL to just always set `valid_user`, and everything works. So either there's an issue in the RTL or there's something else I should be doing to ensure `valid_user` gets set.

Reply:
Resolved in https://github.com/chipsalliance/caliptra-rtl/issues/84